node/rpc/src/lib.rs

```diff
@@ -81,6 +81,7 @@ where
                 chain_head_sub::<DB, KS, B>,
             )
             .with_method("Filecoin.ChainNotify", chain_notify::<DB, KS, B>)
+            .with_method("xrpc.ch.val", chain_notify::<DB, KS, B>)
```

Comment:
Not sure why you need this.

Reply:
You're right. I was experimenting with a different way to handle ChainNotify, but then I realized, the response is formatted like a request. So, I have to manually build the response struct.

I'll take this out.

Reply:
It panicked btw, when i took it out, so there must be a use for it somewhere? No idea xD. It's just really confusing, and I'm ok with you leaving it, but it'd be nice to have a comment to describe what its for.

Reply:
To remove it there's more to refactor. I've done so.